wfsim/strax_interface.py

```diff
@@ -160,6 +177,168 @@ def instruction_from_csv(filename):
+        timings.sort()
+
+        #For tpc we have multiple instructions per g4id. For nveto we only have one
+        counter=Counter(self.instructions_epix['g4id'])
+        i_per_id = [counter[k] for k in range(start,stop)]
+        timings_tpc=np.repeat(timings,i_per_id)
+        self.instructions_epix['time']+=timings_tpc
+        if self.context.config['neutron_veto']:
+            self.instructions_nveto['time']=timings[self.instructions_nveto['g4id']]
+    
+    def set_configuration(self,):
+        '''Set chunking configuration and feeds instructions to wfsim'''
+        max_events_per_chunk=500#is this hard coding really nessecairy...
+        chunk_size = np.ceil(max_events_per_chunk/self.context.config['event_rate'])
+        nchunk = np.ceil((self.context.config['event_stop']-self.context.config['event_start'])/ \
+                            (chunk_size*self.context.config['event_rate']))
+        self.context.set_config(dict(chunk_size=chunk_size, nchunk=nchunk,))
+        self.context.set_config(dict(wfsim_instructions=self.instructions_epix,))
+        
+        if self.context.config['neutron_veto']:
+            self.context.set_config(dict(wfsim_nveto_instructions=self.instructions_nveto,
+                                         wfsim_nveto_channels=self.nveto_channels,
+                                         wfsim_nveto_timings=self.nveto_timings))
+
+    def set_instructions(self,):
+        '''Gets instructions from epix and neutron veto if needed. 
+        Afterwards sets the correct timings and configuration'''
+        self.instructions_from_epix()
+        if self.context.config['neutron_veto']:
+            self.instructions_from_nveto()
+        self.set_timing()
+        self.set_configuration()
+
+    def run_strax(self,run_id):
+        '''Runs wfsim up to raw records for tpc and if requisted the nveto.
+            All this setting neutron_veto=True/False is needed to get wfsim to do the right thing
+        '''
+        self.context.set_config(dict(neutron_veto=False))
+        self.context.make(run_id,'raw_records')
+
+        #nveto doesn't have proper afterpulses or noise implementation
+        #Cannot use neutron_veto=True/False here
+        if len(self.targets)>1:
+            self.context.set_config(dict(neutron_veto=True,))
+            self.context.make(run_id,'raw_records_nv')
+        
+    def _sync_meta(self,):
+        '''Sync metadata between tpc and nveto. The start and end times of the metadata must match for
+            something strax wants. Check which ever one starts first, make that the start of both. Same for end'''
+        logging.debug('Syncing metadata')
+        keys = ('start','end')
+        for key in keys:
+            if self.tpc_meta[key]>self.nveto_meta[key]:
+                self.tpc_meta[key]=self.nveto_meta[key]
+            else:
+                self.nveto_meta[key]=self.tpc_meta[key]
```

Comment:
Also be careful you also have to updated the chunk start of the first chunk and the chunk end of the last chunk.